src/machine/machine_stm32f103xx.go

```diff
@@ -32,6 +33,20 @@ const (
 	PinOutputModeAltOpenDrain PinMode = 12 // Output mode alt. purpose open drain
 )
 
+// Configure this pin with the given I/O settings.
+// stm32f1xx uses different technique for setting the GPIO pins than the stm32f407
+func (p Pin) configure(config PinConfig) {
+	// Configure the GPIO pin.
+	port := p.getPort()
+	pin := uint8(p) % 16
+	pos := uint8(p) % 8 * 4
+	if pin < 8 {
+		port.CRL.SetMasked(uint32(config.Mode), 0xf, pos)
+	} else {
```

Comment:
pos -= 8

Reply:
see note above

Reply:
it looks like the f103 code was doing it right (line 42), but it seems the f407 code had the error I propagated. Thanks again for the catch!